python/grass/jupyter/testsuite/display_test.py

```diff
@@ -0,0 +1,155 @@
+        # Add files to self for cleanup later
+        self.files.append(custom_filename)
+        self.files.append(f"{custom_filename}.grass_vector_legend")
+        # Add a vector and a raster to the map
+        grass_renderer.run("d.rast", map="elevation")
+        grass_renderer.run("d.vect", map="roadsmajor")
+        # Make sure image was created
+        self.assertFileExists(custom_filename)
+
+    @unittest.skipIf(not can_import_folium(), "Cannot import folium")
+    def test_hw(self):
+        """Test that GrassRenderer creates maps with custom height and widths."""
+        # Create map with height and width parameters
+        grass_renderer = gj.GrassRenderer(width=400, height=400)
+        # Add just a vector (for variety here)
+        grass_renderer.run("d.vect", map="roadsmajor")
+
+    @unittest.skipIf(not can_import_folium(), "Cannot import folium")
+    def test_env(self):
+        """Test that we can hand an environment to GrassRenderer."""
+        # Create map with environment parameter
+        grass_renderer = gj.GrassRenderer(env=os.environ.copy())
+        # Add just a raster (again for variety)
+        grass_renderer.run("d.rast", map="elevation")
+
+    @unittest.skipIf(not can_import_folium(), "Cannot import folium")
+    def test_text(self):
+        """Test that we can set a unique text_size in GrassRenderer."""
+        # Create map with unique text_size parameter
+        grass_renderer = gj.GrassRenderer(text_size=10)
+        # Add a vector and a raster
+        grass_renderer.run("d.vect", map="roadsmajor")
+        grass_renderer.run("d.rast", map="elevation")
+
+    @unittest.skipIf(not can_import_folium(), "Cannot import folium")
+    def test_shortcut(self):
+        """Test that we can use display shortcuts with __getattr__."""
+        # Create map
+        grass_renderer = gj.GrassRenderer()
+        # Add raster to map with shortcut
+        grass_renderer.d_rast(map="elevation")
+        grass_renderer.d_vect(map="roadsmajor")
+
+    @unittest.skipIf(not can_import_folium(), "Cannot import folium")
+    def test_shortcut_error(self):
+        """Test that passing an incorrect attribute raises
+        appropriate error"""
+        # Create map
+        grass_renderer = gj.GrassRenderer()
+        # Pass bad shortcuts
+        with self.assertRaisesRegex(AttributeError, "Module must begin with 'd_'"):
+            grass_renderer.r_watersheds()
+        with self.assertRaisesRegex(AttributeError, "d.module.does.not.exist"):
+            grass_renderer.d_module_does_not_exist()
+
+    @unittest.skipIf(not can_import_folium(), "Cannot import folium")
+    def test_InteractiveMap(self):
+        # Create InteractiveMap
+        interactive_map = gj.InteractiveMap()
+        # Add raster and vector
+        interactive_map.add_raster("elevation")
+        interactive_map.add_vector("roadsmajor")
+        # Show
+        interactive_map.show()
```

Comment:
In the command line, `interactive_map.show()` returns a folium object (specifically, something like `branca.element.Figure object at <...>`. I added a comment to the `show()` docstring saying it returned  a folium figure object. I'm not sure if this is problematic or not - my guess is not because the object is just taking up Python memory and will be deleted after Python is exited?

Reply:
That sounds good. `show()` here is little surprising as in e.g. Matplotlib it may mean showing a window and waiting for a user to close it, i.e., show() seems like something will be shown (not the case in the tests), rather than an independent (although perhaps useless) object being returned. It makes sense in the context of Jupyter notebook, so let's leave it as is.